docs/contributing/Regular-Expression-Regex.html

```diff
@@ -0,0 +1,1140 @@
+
+<pre><code>  - name: 'AcolyteMasks\.es(l|p)'
+    url:
+      - link: 'https://www.nexusmods.com/skyrimspecialedition/mods/41713/'
+        name: 'DragonPriest Retexture'
+</code></pre>
+
+<p>Three different aspects have been changed respectively added in order for this regular expression to work.</p>
+
+<ul>
+  <li>The period <code>.</code> has been escaped through writing a backslash <code>\</code> in front of it. This is necessary so that the period is actually recognized as such. Escaping means reducing ambiguity in a more general sense.</li>
+  <li>After the string <code>es</code> follows a capturing group <code>( )</code>. The content of this capturing group needs to follow what is infront of it.</li>
+  <li>Inside the capturing group are the letters l and p, seperated by the OR operand <code>|</code>. Both letters are valid alternatives for our regex.</li>
+</ul>
+
+<p>Both plugins <code>AcolyteMasks.esl</code> and <code>AcolyteMasks.esp</code> will now have their locations displayed in LOOT.</p>
+
+
+<h2>Characters and Rules</h2>
+
+<p>The following regular expressions can be used to match characters:</p>
+<p><code>\d \D \w \W \s \S</code></p>
+
+<p>In order to get a feeling what sorts of characters these expressions are capable of matching, let's take the following set of characters as reference:</p>
+
+<p>Numbers:</p>
+<p><code>0 1 2 3 4 5 6 7 8 9</code></p>
+
+<p>Latin Alphabet:</p>
+<p><code>A B C D E F G H I J K L M N O P Q R S T U V W X Y Z</code></p>
+
+<p>Additional Latin Characters:</p>
+<p><code>Ä Æ Ö Ü</code></p>
+
+<p>Special Characters:</p>
+<p><code>! # $ % & ( ) , . ' ` - ; [ ] ^ _ { } ~ € + = Œ ੩</code></p>
+
+<p>Greek Alphabet:</p>
+<p><code>α β γ δ ε ζ η θ ι κ λ μ ν ξ ο π ρ σ ς τ υ φ χ ψ ω</code></p>
+
+<p>Japanese Hiragana:</p>
+<p><code>あ い う え お か き く け こ さ し す せ そ た ち つ て と な に ぬ ね の は ひ ふ へ ほ ま み む め も や ゆ よ ら り る れ ろ わ を ん</code></p>
+
+<p>Japanese Katakana:</p>
+<p><code>ア イ ウ エ オ カ キ ク ケ コ サ シ ス セ ソ タ チ ツ テ ト ナ ニ ヌ ネ ノ ハ ヒ フ ヘ ホ マ ミ ム メ モ ヤ ユ ヨ ラ リ ル レ ロ ワ ヲ ン</code></p>
+
+<p>Through testing the following rules could be derived:</p>
+
+<ul>
+  <li><code>\d</code> can be used exclusively for integer numbers 0 to 9</li>
+  <li><code>\D</code> can be used to identify (non-number) characters from the Latin Alphabet (except <code>Ä Æ Ö Ü</code>) and all the Special Characters</li>
+
+  <li><code>\w</code> can be used for the integer numbers 0 to 9, for any of the normal letters from the Latin Alphabet and for the underscore _</li>
+  <li><code>\W</code> can detect exclusively all the special characters except the underscore _ (and no numbers or Latin characters)</li>
+
+  <li><code>\s</code> detects whitespace</li>
+  <li><code>\S</code> can detect non-whitespace characters: Numbers, Latin characters (except additional ones) and the special characters</li>
+
+  <li>None of the expressions <code>\d \D \w \W \S</code> apply for the Greek and Japanese characters</li>
```

Comment:
What characters are recognised varies by regex implementation used, so I think that these statements should be qualified to make it clear that they're talking about LOOT's use of regex in the name field of plugin objects in the masterlist.

Even then, LOOT uses three different regex implementations in different places, and this means:

* The search dialog uses Qt's regex implementation, which doesn't support unicode characters by default (but it can, I just didn't set the flag).
* The contains filter and masterlist entry matching uses C++'s standard regex, which doesn't support unicode.
* Condition evaluation uses the Rust regex library, which supports unicode by default.